dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.service.description.3.2/src/org/apache/cxf/jaxrs/model/wadl/WadlGenerator.java

```diff
@@ -0,0 +1,2333 @@
+    private static final String DEFAULT_NS_PREFIX = "prefix";
+    private static final Map<ParameterType, Class<? extends Annotation>> PARAMETER_TYPE_MAP;
+    static {
+        PARAMETER_TYPE_MAP = new HashMap<>();
+        PARAMETER_TYPE_MAP.put(ParameterType.FORM, FormParam.class);
+        PARAMETER_TYPE_MAP.put(ParameterType.QUERY, QueryParam.class);
+        PARAMETER_TYPE_MAP.put(ParameterType.HEADER, HeaderParam.class);
+        PARAMETER_TYPE_MAP.put(ParameterType.PATH, PathParam.class);
+        PARAMETER_TYPE_MAP.put(ParameterType.MATRIX, MatrixParam.class);
+    }
+
+    private String wadlNamespace;
+
+    private boolean singleResourceMultipleMethods = true;
+    private boolean useSingleSlashResource;
+    private boolean ignoreForwardSlash;
+    private boolean linkAnyMediaTypeToXmlSchema;
+    private boolean useJaxbContextForQnames = true;
+    private boolean supportCollections = true;
+    private boolean supportJaxbXmlType = true;
+    private boolean supportJaxbSubstitutions = true;
+    private boolean ignoreOverloadedMethods;
+    private boolean checkAbsolutePathSlash;
+    private boolean keepRelativeDocLinks;
+    private boolean usePathParamsToCompareOperations = true;
+    private boolean incrementNamespacePrefix = true;
+
+    private boolean ignoreMessageWriters = true;
+    private boolean ignoreRequests;
+    private boolean convertResourcesToDOM = true;
+    private String wadlSchemaLocation;
+    private List<String> externalSchemasCache;
+    private List<URI> externalSchemaLinks;
+    private Map<String, List<String>> externalQnamesMap;
+
+    private final ConcurrentHashMap<String, String> docLocationMap = new ConcurrentHashMap<>();
+
+    private ElementQNameResolver resolver;
+    private List<String> privateAddresses;
+    private List<String> whiteList;
+    private String applicationTitle;
+    private String nsPrefix = DEFAULT_NS_PREFIX;
+    private MediaType defaultWadlResponseMediaType = MediaType.APPLICATION_XML_TYPE;
+    private final MediaType defaultRepMediaType = MediaType.WILDCARD_TYPE;
+    private String stylesheetReference;
+    private boolean applyStylesheetLocally;
+    private Bus bus;
+    private final List<DocumentationProvider> docProviders = new LinkedList<>();
+    private ResourceIdGenerator idGenerator;
+    private Map<String, Object> jaxbContextProperties;
+
+    public WadlGenerator() {
+    }
+
+    public WadlGenerator(Bus bus) {
+        this.bus = bus;
+        this.bus.setProperty("wadl.service.description.available", "true");
+    }
+
+    @Override
+    public void filter(ContainerRequestContext context) {
+        Message m = JAXRSUtils.getCurrentMessage();
+        if (m == null) {
+            return;
+        }
+        doFilter(context, m);
+    }
+
+    protected void doFilter(ContainerRequestContext context, Message m) {
+        //Liberty code change start
+        if (!"GET".equals(((MessageImpl) m).getHttpRequestMethod())) {
```

Comment:
If I read this right, we are pulling in the WadlGenerator source file and updating the bnd file just for this change.  I don't think very many customers use Wadl (it's pretty much been replaced by Swagger/OpenAPI).  Are you seeing specific performance improvements in your benchmarks with this change?  If so, I'd probably update the benchmarks to use OpenAPI instead. :) If not, I'd say let's undo this change and the bnd change - it's one less divergence from the Apache CXF repo.

Reply:
What API should I look for in my app to determine what's bringing in the WadlGenerator? I'm not explicitly using wadl (just pinging a simple REST endpoint). The only reason I changed that code is because it's showing up in the profiles as a caller into MessageImpl.get. JAXRSUtils.runContainerRequestFilters -> WadlGenerator.filter -> WadlGenerator.doFilter is the path I observe

Reply:
That's unfortunate - it looks like we always load and register that provider... In the base CXF code, they only load it if the service.description module is on the class path - and we have it on the class path, so that means that we're executing that filter on every request...

To remove it now would probably break zero-migration - so we should probably fix it to be faster if possible.  Thanks